tests/test_dialects.py

```diff
@@ -991,6 +1217,77 @@ def test_hive(self):
         self.validate(
             "CAST(1 AS INT)", "CAST(1 AS INT)", read="hive", write="starrocks"
         )
+        self.validate(
+            "MIXED_TYPE_TO_DATE_STR(x)",
+            "DATE_FORMAT(TO_DATE(SUBSTR(REPLACE(CAST(x as string), '-', ''), 1, 8), 'yyyyMMdd'), 'yyyy-MM-dd')",
+            write="hive",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_TO_DATE(x)",
+            "TO_DATE(SUBSTR(REPLACE(CAST(x as string), '-', ''), 1, 8), 'yyyyMMdd')",
```

Comment:
is this correct? shouldn't mixed_type_to_date(x) in hive just be TO_DATE(x)

Reply:
Hive does not support int as a type for the `TO_DATE` function. So we need to first get it into a string in the format that we are providing.